packages/commonwealth/scripts/async-migrate-community-count.ts

```diff
@@ -0,0 +1,48 @@
+import { dispose } from '@hicommonwealth/core';
+import { models } from '@hicommonwealth/model';
+import { QueryTypes } from 'sequelize';
+
+async function run() {
+  try {
+    const communityIds = (
+      await models.sequelize.query<{ id: string }>(
+        `SELECT id FROM "Communities" where count_updated = false`,
+        {
+          raw: true,
+          type: QueryTypes.SELECT,
+        },
+      )
+    ).map((c) => c.id);
+
+    for (const community_id of communityIds) {
+      console.log(`Updating thread/profile counts for ${community_id}`);
+      await models.sequelize.query(
+        `
+        UPDATE "Communities"
+        SET count_updated = true,
+            thread_count = (SELECT COUNT(*) FROM "Threads" WHERE community_id = :community_id),
+            profile_count = (
+                SELECT COUNT(DISTINCT p.id)
+                FROM "Profiles" p
+                JOIN "Addresses" a ON p.id = a.profile_id
+                WHERE a.community_id = :community_id
+            )
+        WHERE id = :community_id;
+        `,
+        { replacements: { community_id } },
+      );
```

Comment:
Considering this is technically a migration, should these happen in a transaction?

Reply:
Probably not, there is no good way to hold the transaction here as it will lock the tables. This script is meant to be run async anyways and can be run multiple times, so there should be no data consistency issues.

Reply:
Ccing @jnaviask & @ilijabojanovic for visibility here.